tests/helpers/app-fixture.js

```diff
@@ -0,0 +1,253 @@
+
+function AppFixture(name) {
+  this.type = 'app';
+  this.command = 'new';
+  this.name = name;
+  this._installedAddons = [];
+
+  this._init();
+}
+
+AppFixture.prototype = {
+
+  _init: function() {
+    process.chdir(root);
+    this.dir = quickTemp.makeOrRemake({}, this.name + '-' + this.type + '-fixture');
+    process.chdir(originalWorkingDirectory);
+
+    this._loadBlueprint();
+  },
+
+  _loadBlueprint: function() {
+    fs.removeSync(this.dir);
+
+    ember.invoke(
+      this.command,
+      this.name,
+      '--directory=' + this.dir,
+      '--disable-analytics',
+      '--watcher=node',
+      '--skip-npm',
+      '--skip-bower',
+      '--skip-git'
+    );
+
+    this.fixture = fixturify.readSync(this.dir);
+
+    // Clean up after the generator.
+    fs.removeSync(this.dir);
+  },
+
+  serialize: function(isChild) {
+    var npmLinks = [];
+    var inRepoLinks = [];
+    var self = this;
+    this._installedAddons.forEach(function(addon) {
+      addon.serialize(true);
+
+      if (addon.type === 'addon') {
+        npmLinks.push({
+          name: addon.name,
+          path: addon.dir
+        });
+      } else if (addon.type === 'in-repo-addon') {
+        inRepoLinks.push({
+          from: path.join(self.dir, 'lib', addon.name),
+          to: addon.dir
+        });
+      }
+    });
+
+    fixturify.writeSync(this.dir, this.fixture);
+
+    var packageCache = new PackageCache(root);
+
+    var from, to;
+    if (this.fixture['package.json'] || npmLinks.length) {
+      var nodePackageCache;
+      if (isChild) {
+        process.env.NODE_ENV = 'production';
+        nodePackageCache = packageCache.create(this.type + '-production-node', 'yarn', this.fixture['package.json'], npmLinks);
+        delete process.env.NODE_ENV;
+      } else {
+        nodePackageCache = packageCache.create(this.type + '-node', 'yarn', this.fixture['package.json'], npmLinks);
+      }
+
+      from = path.join(nodePackageCache, 'node_modules');
+      fs.mkdirsSync(from); // Just in case the path doesn't exist.
+      to = path.join(this.dir, 'node_modules');
+      symlinkOrCopySync(from, to);
+    }
+
+    if (!isChild && this.fixture['bower.json']) {
+      var bowerPackageCache = packageCache.create(this.type + '-bower', 'bower', this.fixture['bower.json']);
+
+      from = path.join(bowerPackageCache, 'bower_components');
+      fs.mkdirsSync(from); // Just in case the path doesn't exist.
+      to = path.join(this.dir, 'bower_components');
+      symlinkOrCopySync(from, to);
+    }
+
+    inRepoLinks.forEach(function(link) {
+      fs.mkdirsSync(path.dirname(link.from)); // Just in case the path doesn't exist.
+      fs.mkdirsSync(path.dirname(link.to)); // Just in case the path doesn't exist.
+      symlinkOrCopySync(link.to, link.from);
+    });
+
+    return this;
+  },
+
+  clean: function() {
+    this._installedAddons.forEach(function(addon) {
+      addon.clean(true);
+    });
+
+    // Build up object to pass to quickTemp.
```

Comment:
`quickTemp` falls apart when you need a singular directory since its API is designed for making many, many directories. Trying to keep the directory creation and deletion logic wrapped in one bit of code. At this point I bet we're the sole user of that lib, I nominate fork and improve its API, not copypasta code.